PyInstaller/lib/modulegraph/modulegraph.py

```diff
@@ -3091,7 +3092,14 @@ def _find_module_path(self, fullname, module_name, search_dirs):
                 # Else, this is either a module or C extension.
                 else:
                     # In either case, this path must have a filetype.
-                    filetype = os.path.splitext(pathname)[1]
+                    # Check if pathname ends with a filetype from
+                    # _IMPORTABLE_FILETYPE_TO_METADATA.
+                    # Reverse sort _IMPORTABLE_FILETYPE_TO_METADATA so longest
+                    # filetype matches first.
+                    # Return the filetype or False if none match.
```

Comment:
We need to match more than just the file extension for some file types so `os.path.splitext` won't work in those cases, i.e., `_struct-cpython-m37.dll` from MINGW64, since we would only get `.dll` from `splitext` which doesn't match any valid file types in Python 3 from MINGW64. We need to match the longest filename since some file types are substrings of others, i.e., `.py` and `.pyc`.